gemd/entity/object/tests/test_ingredient_run.py

```diff
@@ -35,3 +35,42 @@ def test_invalid_assignment():
         IngredientRun(process="process")
     with pytest.raises(TypeError):
         IngredientRun(spec=5)
+    with pytest.raises(TypeError):
+        IngredientRun(name="Flour")  # IngredientRuns don't have their own name
+
+
+def test_name_persistance():
+    """Verify that a serialized IngredientRun doesn't lose its name."""
+    from gemd.entity.object import IngredientSpec
+    from gemd.entity.link_by_uid import LinkByUID
+    from gemd.json import GEMDJson
+
+    je = GEMDJson()
+
+    ms_link = LinkByUID(scope='local', id='mat_spec')
+    mr_link = LinkByUID(scope='local', id='mat_run')
+    ps_link = LinkByUID(scope='local', id='pro_spec')
+    pr_link = LinkByUID(scope='local', id='pro_run')
+    spec = IngredientSpec(name='Ingred', labels=['some', 'words'],
+                          process=ps_link, material=ms_link)
+    run = IngredientRun(spec=spec,
+                        process=pr_link, material=mr_link)
```

Comment:
I think you should add here an assertion that `run.name` and `run.labels` match the `spec` values. Otherwise the test below would pass in a situation where ingredient runs did not have any name/label related information.